web/src/main/java/com/epam/rft/atsy/web/controllers/NewApplicationPopupController.java

```diff
@@ -30,20 +39,33 @@ public ModelAndView loadPage() {
   }
 
   @RequestMapping(method = RequestMethod.POST, value = "/secure/new_application_popup")
-  public String saveOrUpdate(@Valid @ModelAttribute StateDTO stateDTO, BindingResult result) {
-    if (!result.hasErrors()) {
-      stateDTO.setStateType("newstate");
-      stateDTO.setStateIndex(0);
-
-      ApplicationDTO applicationDTO = ApplicationDTO.builder()
-          .creationDate(new Date())
-          .candidateId(stateDTO.getCandidateId())
-          .positionId(stateDTO.getPosition().getId())
-          .channelId(stateDTO.getChannel().getId())
-          .build();
-
-      applicationsService.saveApplicaton(applicationDTO, stateDTO);
+  public Object saveOrUpdate(@Valid @ModelAttribute StateDTO stateDTO, BindingResult result, Locale locale) {
+    if (result.hasErrors()) {
```

Comment:
The current implementation was suggested because we thought that the request is an AJAX request and we should return a JSON response.
As it turned out it is not the case, so please revert the controller implementation to the original one.
We will handle this in a separate ticket.